scripts/graphql_api_content/render_helpers.rb

```diff
@@ -10,14 +10,17 @@ def render_html(markdown)
     markdown ? CommonMarker.render_html(markdown) : nil
   end
 
-  def render_of_type(of_type, size = "medium")
-    if of_type["ofType"]
-      render_of_type(of_type["ofType"])
+  def render_of_type(type, nullable = true, size = "medium")
+    if type["ofType"]
+      nullable = type["kind"] != "NON_NULL"
+      render_of_type(type["ofType"], nullable)
     else
-      kind = of_type["kind"] || ""
-      if name = of_type["name"]
+      kind = type["kind"] || ""
+      nullable_symbol = nullable ? "!" : ""
+
+      if name = type["name"]
         <<~HTML
-          <a href="/docs/apis/graphql/schemas/#{kind.downcase}/#{name.downcase}" class="pill pill--#{kind.downcase} pill--normal-case pill--#{size}" title="Go to #{kind} #{name}"><code>#{name}</code></a>
+          <a href="/docs/apis/graphql/schemas/#{kind.downcase}/#{name.downcase}" class="pill pill--#{kind.downcase} pill--normal-case pill--#{size}" title="Go to #{kind} #{name}"><code>#{nullable_symbol}#{name}</code></a>
```

Comment:
The nullable symbol should be a suffix rather than a prefix. From [the spec](https://spec.graphql.org/October2021/#sec-Non-Null):

> A trailing exclamation mark is used to denote a field that uses a Non-Null type like this: `name: String!`.

Reply:
🤦‍♂️!